test/java/controllers/AlertControllerTest.java

```diff
@@ -0,0 +1,236 @@
+import com.arpnetworking.metrics.portal.alerts.AlertRepository;
+import com.arpnetworking.metrics.portal.alerts.impl.PluggableAlertRepository;
+import com.arpnetworking.metrics.portal.integration.test.WebServerHelper;
+import com.arpnetworking.metrics.portal.organizations.OrganizationRepository;
+import com.arpnetworking.metrics.portal.organizations.impl.DefaultOrganizationRepository;
+import com.arpnetworking.metrics.portal.scheduling.impl.MapJobExecutionRepository;
+import com.arpnetworking.testing.SerializationTestUtils;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import models.internal.MetricsQueryFormat;
+import models.internal.Organization;
+import models.internal.QueryResult;
+import models.internal.alerts.Alert;
+import models.internal.alerts.AlertEvaluationResult;
+import models.internal.impl.DefaultAlert;
+import models.internal.impl.DefaultAlertEvaluationResult;
+import models.internal.impl.DefaultMetricsQuery;
+import models.view.alerts.AlertFiringState;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import play.mvc.Result;
+import play.test.Helpers;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Unit tests for {@code AlertController}.
+ * <p>
+ * This should be refactored into an integration/functional test once we can expose an API for alert creation.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+@RunWith(Enclosed.class)
+public final class AlertControllerTest {
+    private static final Instant LAST_INTERVAL = Instant.now();
+
+    private static final List<UUID> FIRING_IDS = Stream.of(
+            "1e7acce3-cd88-47be-8312-c61ccf88ba07",
+            "d2b1ba49-0819-498c-8089-08b8ff15b5bc",
+            "d98798e1-b50d-47c9-ae34-a6aee074dab4",
+            "d4d4f795-55be-4a2a-8d71-22dc2eb237c2",
+            "bf560372-a853-4a9c-948c-9fe9f0269ad7"
+    ).map(UUID::fromString).collect(ImmutableList.toImmutableList());
+
+    private static final List<UUID> NOT_FIRING_IDS = Stream.of(
+            "c6e274d9-d8c6-40e3-aea8-d1146c813c0f",
+            "20adb3d5-4c0a-4b60-a4c1-819a86798bb1",
+            "11dead9e-e5cc-427d-a3a5-381259353b4d",
+            "09d7212f-06b8-4b91-a6f9-384350190cfc",
+            "ce7347cd-9d05-44d1-ac36-f868a14b7c49"
+    ).map(UUID::fromString).collect(ImmutableList.toImmutableList());
```

Comment:
This setup has more IDs than necessary because they will be used in the pagination test. I extracted this PR from it and I left this in to spare myself the headache of needing to rebase again later.